### PR TITLE
OKD (wls-srg) specific changes to integration tests in release/3.3

### PR DESCRIPTION
**Pull request by sankarpn**
release/3.3 integration tests was not supporting OKD platform. These changes are made to run the integration tests in release/3.3 to run on OKD platform.

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9855/testReport/ (all tests passed in release/3.3)
https://build.weblogick8s.org:8443/job/wko-okd-test/249/testReport/

### CODE REVIEW

**Review by rjeberhard — APPROVED** [review]
Approved, but you need to merge conflicts with latest changes.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/WitParams.java [hunk 1]
```diff
@@ -58,6 +58,9 @@
   // Custom WDT model home
   private String wdtModelHome;
 
+  // Custom WDT model home
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/WitParams.java:61]: delete or change this comment line

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Domain.java [hunk 2]
```diff
@@ -186,27 +171,30 @@ public static boolean adminNodePortAccessible(int nodePort, String userName, Str
...
+    String hostAndPort = getHostAndPort(routeHost[0], nodePort);
     String consoleUrl = new StringBuffer()
         .append("http://")
         .append(hostAndPort)
         .append("/console/login/LoginForm.jsp").toString();
 
-    getLogger().info("Accessing WebLogic console with url {0}", consoleUrl);
+    logger.info("Accessing WebLogic console with url {0}", consoleUrl);
     final WebClient webClient = new WebClient();
+    //final HtmlPage loginPage = assertDoesNotThrow(() -> webClient.getPage(consoleUrl),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Domain.java:187]: delete this commented out line

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java [hunk 3]
```diff
@@ -521,27 +522,18 @@ private static void createDomainJob(String pvName,
...
+               .name("create-weblogic-domain-onpv-container")
                         .image(WEBLOGIC_IMAGE_TO_USE_IN_SPEC)
                         .imagePullPolicy("IfNotPresent")
                         .addPortsItem(new V1ContainerPort()
                             .containerPort(7001))
                         .volumeMounts(Arrays.asList(
                             new V1VolumeMount()
                                 .name("create-weblogic-domain-job-cm-volume") // domain creation scripts volume
-                                .mountPath("/u01/weblogic"), // availble under /u01/weblogic inside pod
+                                  .mountPath("/u01/weblogic"), // availble under /u01/weblogic inside pod
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java:536]: .name() and .mountPath() should be in the same level, reformat it

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FmwUtils.java [hunk 4]
```diff
@@ -218,4 +220,34 @@ public static void verifyDomainReady(String domainNamespace, String domainUid, i
     assertTrue(callWebAppAndWaitTillReady(curlCmd1, 5), "Calling web app failed");
     logger.info("EM console is accessible thru default service");
   }
-}
\ No newline at end of file
+
+  /**
+   * Verify EM console is accessible.
+   * @param domainUid unique Uid of the domain
+   * @param domainNamespace  namespace where the domain exists
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FmwUtils.java:227]: The parameter order is wrong. domainNamespace should be the first one.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/PersistentVolumeUtils.java [hunk 5]
```diff
@@ -207,9 +215,28 @@ public static void createPV(String pvName, String domainUid, String className) {
...
-        "Failed to create persistent volume");
-    assertTrue(success, "PersistentVolume creation failed");
+  }
+
+  @NotNull
+  private static Path createPVHostPathDir(String pvName, String className) {
+    Path pvHostPath = null;
+    LoggingFacade logger = getLogger();
+    try {
+      pvHostPath = Files.createDirectories(Paths.get(PV_ROOT, className, pvName));
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/PersistentVolumeUtils.java:225]: use static import Files.createDirectories and Paths.get here

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/PersistentVolumeUtils.java [hunk 6]
```diff
@@ -283,6 +310,77 @@ public static synchronized V1Container createfixPVCOwnerContainer(String pvName,
...
+      hasLabels = true;
+      v1pv.getMetadata().setLabels(labels);
+      labelSelector = labels.entrySet()
+          .stream()
+          .map(e -> e.getKey() + "="
+              + e.getValue())
+          .collect(Collectors.joining(","));
+    }
+
+
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/PersistentVolumeUtils.java:357]: delete empty line

**Review by xiancao — CHANGES_REQUESTED** [review]

**Review by maggiehe00 — APPROVED** [review]
I'm approving this PR since I know all the single test passed on another OKD cluster. Let's get it merge and we can fix/improve later.

**Comment by rjeberhard** [comment]
Closing `release/3.3` PR's.

**Pull request closed without merging**